homeassistant/components/habitica/sensor.py

```diff
@@ -166,128 +172,111 @@ class HabitipySensorEntity(StrEnum):
-        for element in self.entity_description.value_path:
-            data = data[element]
-        return cast(StateType, data)
 
+        return self.entity_description.value_fn(self.coordinator.data.user)
 
-class HabitipyTaskSensor(
-    CoordinatorEntity[HabiticaDataUpdateCoordinator], SensorEntity
-):
-    """A Habitica task sensor."""
 
-    def __init__(self, name, task_name, coordinator, entry):
-        """Initialize a generic Habitica task."""
-        super().__init__(coordinator)
-        self._name = name
-        self._task_name = task_name
-        self._task_type = TASKS_TYPES[task_name]
-        self._state = None
-        self._attr_unique_id = f"{entry.unique_id}_{task_name}"
-        self._attr_device_info = DeviceInfo(
-            entry_type=DeviceEntryType.SERVICE,
-            manufacturer=MANUFACTURER,
-            model=NAME,
-            name=entry.data[CONF_NAME],
-            configuration_url=entry.data[CONF_URL],
-            identifiers={(DOMAIN, entry.unique_id)},
-        )
+class HabitipyTaskSensor(HabiticaBase, SensorEntity):
+    """A Habitica task sensor."""
 
-    @property
-    def icon(self):
-        """Return the icon to use in the frontend, if any."""
-        return self._task_type.icon
+    entity_description: HabitipyTaskSensorEntityDescription
 
     @property
-    def name(self):
-        """Return the name of the task."""
-        return f"{DOMAIN}_{self._name}_{self._task_name}"
-
-    @property
-    def native_value(self):
+    def native_value(self) -> StateType:
         """Return the state of the device."""
-        return len(
-            [
-                task
-                for task in self.coordinator.data.tasks
-                if task.get("type") in self._task_type.path
-                and not task.get("completed")
-            ]
-        )
+
+        return len(self.entity_description.value_fn(self.coordinator.data.tasks))
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
         """Return the state attributes of all user tasks."""
         attrs = {}
 
         # Map tasks to TASKS_MAP
-        for received_task in self.coordinator.data.tasks:
-            if received_task.get("type") in self._task_type.path:
-                task_id = received_task[TASKS_MAP_ID]
-                task = {}
-                for map_key, map_value in TASKS_MAP.items():
-                    if value := received_task.get(map_value):
-                        task[map_key] = value
-                attrs[task_id] = task
+        for received_task in self.entity_description.value_fn(
+            self.coordinator.data.tasks
+        ):
+            task_id = received_task[TASKS_MAP_ID]
+            task = {}
+            for map_key, map_value in TASKS_MAP.items():
+                if value := received_task.get(map_value):
+                    task[map_key] = value
+            attrs[task_id] = task
         return attrs
```

Comment:
Can we deprecate this in a follow up now since we have a todo now?

Reply:
The todo and daily sensors are already marked as deprecated, a repair issue will show up if they are used in a automation or a script. Or should I do something else?

Reply:
Oh right, I thought we only deprecated a part of it

Reply:
We still have the rewards and habits sensors, they aren't deprecated yet, cause there is no replacement

Reply:
and they don't have these extra state attributes I made this comment to?

Reply:
yea, they have 😬
But how should i deprecate them? Mark the whole sensors like for the todo and dailies or remove the extra attributes only?

Reply:
Let's leave this for a follow up, but we should make sure we can fetch the data via service calls and just deprecate it